expression/constant.go

```diff
@@ -450,3 +451,62 @@ func (c *Constant) Coercibility() Coercibility {
 	}
 	return c.collationInfo.Coercibility()
 }
+
+// TryUpdateCollate Update collate based on priority #35149
+func TryUpdateCollate(eq *ScalarFunction) {
+
+	if eq.FuncName.L != ast.EQ {
+		return
+	}
+	ok := expressionUpdateCollate(eq, true)
```

Comment:
you can simplify as one line: expressionUpdateCollate(eq, true) and remove the second pram of expressionUpdateCollate

Reply:
done